aten/src/ATen/core/Dict_inl.h

```diff
@@ -19,6 +23,27 @@ inline bool shallowEquals(const IValue& lhs, const IValue& rhs) {
     AT_ERROR("shallowEquals(IValue, IValue) not implemented for type ", lhs.tagKind());
   }
 }
+
+template<class Key, class Value>
+Dict<Key, Value> toTypedDict(GenericDict dict) {
+  if (dict.impl_->elementTypes.has_value()) {
+    // TODO Show types in error message: TORCH_INTERNAL_ASSERT(getTypePtr<Key>() == dict.impl_->elementTypes->keyType, "Tried to cast a Dict with key type ", (*dict.impl_->elementTypes->keyType)->str(), " to a dict with different key type ", getTypePtr<Key>()->str());
+    TORCH_INTERNAL_ASSERT(getTypePtr<Key>() == dict.impl_->elementTypes->keyType, "Tried to cast a Dict to a dict with different key type");
+    // TODO Show types in error message: TORCH_INTERNAL_ASSERT(getTypePtr<Value>() == dict.impl_->elementTypes->valueType, "Tried to cast a Dict with value type ", (*dict.impl_->elementTypes->valueType)->str(), " to a dict with different value type ", getTypePtr<Value>()->str());
+    TORCH_INTERNAL_ASSERT(getTypePtr<Value>() == dict.impl_->elementTypes->valueType, "Tried to cast a Dict to a dict with different value type");
+  } else {
+    // The dict wasn't created with type info, but we're just about to convert it to a concrete type.
+    // Store that type info to the dict so that future conversions are checked against it.
+    dict.impl_->elementTypes = detail::DictImpl::DictElementTypes {getTypePtr<Key>(), getTypePtr<Value>()};
```

Comment:
technically it can be a race condition if multiple ops are called on the same shared dict. Also, mutability of the dict from first toTypedDict access sounds surprising. Do you really need it? (if you plan to phase out generic constructors in the future)

Reply:
I deleted it